javalib/src/main/scala/java/nio/channels/FileChannel.scala

```diff
@@ -113,24 +124,30 @@ object FileChannel {
       mode.append("s")
     }
 
-    val file = path.toFile()
-    val raf = new RandomAccessFile(file, mode.toString)
+    val raf = tryRandomAccessFile(path.toString, mode.toString)
 
-    if (writing && options.contains(TRUNCATE_EXISTING)) {
-      raf.setLength(0L)
-    }
+    try {
+      if (writing && options.contains(TRUNCATE_EXISTING)) {
+        raf.setLength(0L)
+      }
 
-    if (writing && options.contains(APPEND)) {
-      raf.seek(raf.length())
-    }
+      if (writing && options.contains(APPEND)) {
+        raf.seek(raf.length())
+      }
 
-    new FileChannelImpl(
-      raf.getFD(),
-      Some(file),
-      deleteFileOnClose = options.contains(StandardOpenOption.DELETE_ON_CLOSE),
-      openForReading = true,
-      openForWriting = writing
-    )
+      new FileChannelImpl(
+        raf.getFD(),
+        Some(path.toFile()),
+        deleteFileOnClose =
+          options.contains(StandardOpenOption.DELETE_ON_CLOSE),
+        openForReading = true,
+        openForWriting = writing
+      )
+    } catch {
+      case e: Throwable =>
+        raf.close()
```

Comment:
I see your point that `e` is probably what people want to know about and that any
exception from raf.close() should be swallowed.  

Thank you, have a fix on the way (has to pass my local testing before I submit).

This is javalib code, so by my understanding I can not use `scala.util.Try`
and must do it the long, explicit way.

There is no need to check if raf is null because the try block related
to the catch block containing that code is only entered if raf was
non-null at that point.  `tryRandomAccessFile` would have thrown
an Exception if anything there had failed.  

re: close()
We need to close the underlying RAF file on failure (only)
because it is unlikely that anyone will be doing a `FileChannel.close()` after
a `FileChannel.open()` failure.  Implementation details should not be exposed
and Implementations should do a best effort to clean themselves up after errors.